src/lib/videoUtils.ts

```diff
@@ -189,4 +180,32 @@ export const formatDurationToMinutesSeconds = (start: Date, end: Date): string =
 export const extractCameraName = (filename: string): string => {
   const match = filename.match(/video_([^/]+)/);
   return match ? `${match[1]}` : 'Unknown Camera';
-};
\ No newline at end of file
+};
+
+const formatDateToVideoStoreFormat = (date: Date): string => {
+  const year = date.getUTCFullYear();
+  const month = String(date.getUTCMonth() + 1).padStart(2, '0');
+  const day = String(date.getUTCDate()).padStart(2, '0');
+  const hours = String(date.getUTCHours()).padStart(2, '0');
+  const minutes = String(date.getUTCMinutes()).padStart(2, '0');
+  const seconds = String(date.getUTCSeconds()).padStart(2, '0');
+  // Z means UTC
+  return `${year}-${month}-${day}_${hours}-${minutes}-${seconds}Z`;
+};
+
+export const generateVideo = async (
+  videoStoreClient: VIAM.GenericComponentClient, 
+  step: Step) => {
+    console.log("generateVideo called for step", step);
+    console.log("formatDateToVideoStoreFormat(step.end)", formatDateToVideoStoreFormat(step.end));
+    console.log("formatDateToVideoStoreFormat(step.start)", formatDateToVideoStoreFormat(step.start));
+    const command = VIAM.Struct.fromJson({
+      "command": "save",
+      "to": formatDateToVideoStoreFormat(step.end),
+      "from": formatDateToVideoStoreFormat(step.start),
+      "metadata": `${step.pass_id}${step.name}`,
+      "async": true
```

Comment:
don't do async true here

We want to know if it succeeds or fails and report an error if it fails